thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/route.js

```diff
@@ -443,6 +395,98 @@ export default Route.extend({
+    }
+  },
+
+  /**
+   * Fetch all anomalies we have Ids for. Enhance the data and populate power-select filter options.
+   * Using ember concurrency parent/child tasks. When parent is cancelled, so are children
+   * http://ember-concurrency.com/docs/child-tasks.
+   * TODO: complete concurrency task error handling and refactor child tasks for cuncurrency.
+   * @param {Array} anomalyIds - the IDs of anomalies that have been reported for this alert.
+   * @return {undefined}
+   */
+  loadAnomalyData: task(function * (anomalyIds) {
+    yield timeout(300);
+    // Load data for each anomaly Id
+    const rawAnomalies = yield fetchCombinedAnomalies(anomalyIds);
+    // Fetch and append severity score to each anomaly record
+    const severityScores = yield fetchSeverityScores(anomalyIds);
+    // Process anomaly records to make them template-ready
+    const anomalyData = yield enhanceAnomalies(rawAnomalies, severityScores);
+    // Prepare de-duped power-select option arrays
+    resolutionOptions.push(...new Set(anomalyData.map(record => record.anomalyFeedback)));
+    dimensionOptions.push(...new Set(anomalyData.map(anomaly => anomaly.dimensionString)));
+    // Push anomaly data into controller
+    this.controller.setProperties({
+      anomaliesLoaded: true,
+      anomalyData,
+      resolutionOptions,
+      dimensionOptions
+    });
+    // Fetch and append extra WoW data for each anomaly record
+    const wowData = yield this.fetchCombinedAnomalyChangeData(anomalyData);
+    anomalyData.forEach((anomaly) => {
+      anomaly.wowData = wowData[anomaly.anomalyId] || {};
+    });
+    // Load enhanced dataset into controller (WoW options will appear)
+    this.controller.setProperties({
+      anomalyData,
+      baselineOptionsLoading: false,
+      baselineOptions: [baselineOptions[0], ...newWowList]
+    });
+  // We use .cancelOn('deactivate') to make sure the task cancels when the user leaves the route.
+  // We use restartable to ensure that only one instance of the task is running at a time, hence
+  // any time setupController performs the task, any prior instances are canceled.
+  }).cancelOn('deactivate').restartable(),
+
+  /**
+   * Concurrenty task to ping the job-info endpoint to check status of an ongoing replay job.
+   * If there is no progress after a set time, we display an error message.
+   * @param {Number} jobId - the id for the newly triggered replay job
+   * @param {String} functionName - user-provided new function name (used to validate creation)
+   * @return {undefined}
+   */
+  loadGraphData: task(function * (metricDataUrl, exploreDimensions) {
+    yield timeout(300);
+    const metricId = this.currentModel.config ? this.currentModel.config.id : null;
+    const isGraphDataLocal = metricId && localStorage.getItem(metricId) !== null;
+
+    // Fetch and load graph metric data.
+    fetch(metricDataUrl).then(checkStatus)
```

Comment:
Is this what you were referring to? My understanding is that moving this to another async function will allow for `await/async`, but doesn't work in the context of a task generator?

Reply:
Correct. https://emberway.io/ember-concurrency-the-solution-to-so-many-problems-you-never-knew-you-had-cce6d7731ba9

This may be outdated by now, but "until the proposal supports cancelation, ember-concurrency can’t use async functions as task functions."